File: src/main/java/com/example/calendarexperiments/jackson/CustomConfiguration.java
package com.example.calendarexperiments.jackson;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class CustomConfiguration {

	@Bean
	public CustomObjectMapper getObjectMapper() {
		return new CustomObjectMapper();
	}

	public static class CustomObjectMapper extends ObjectMapper {
		public CustomObjectMapper() {
			registerModule(new JavaTimeModule());
		}
	}
}

File: src/main/kotlin/com/example/calendarexperiments/jackson/CustomConfiguration.kt
@file:Suppress("unused")

package com.example.calendarexperiments.jackson

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import org.springframework.context.annotation.Configuration

@Configuration
class CustomConfiguration {

    val objectMapper: CustomObjectMapper
        get() = CustomObjectMapper()

    @Configuration
    class CustomObjectMapper : ObjectMapper() {
        init {
            registerModule(JavaTimeModule())
        }
    }
}